documentation/architecture/decisions/0004-cloudwatch-metrics.md

```diff
@@ -0,0 +1,83 @@
+
+We had a spike to see if we could use the Cloudwatch exporter to get Cloudwatch metrics and trigger alerts. 
+After getting it to work with a number of metrics we were able to start to estimate costs, from these estimate we had concerns regarding the [high costs](https://aws.amazon.com/cloudwatch/pricing/)
+in running it due to the number of metrics requested using the Cloudwatch API.
+
+Each time the `/metrics` endpoint was hit triggers the exporter to retrieve
+metrics from the Cloudwatch API. Every Cloudwatch metric that you request costs
+money (regardless of if you ask for 100 metrics using 1 metric per api call or
+100 metrics using 100 metrics per api call).
+
+cost = `number of metrics on /metrics page` x `number of scrapes an hour` x `number of hours in a year` x `price of requesting a metric using the API` x `number of prometheis running across all our environments`.
+
+Based on a simple assumption of 100 metrics requested, being scraped once every 10 minutes (6 per hour), 3 Prometheis in production, 3 Prometheis in staging and 4 in dev accounts, it would work out:
+
+`100 x 6 x 24 x 365 x $0.00001 x 10 = $525.6 per year`
+
+However if we wish to scrape at the same rate we would a normal target, e.g. 30
+seconds that would become roughly $10,500 per year.
+
+100 metrics also appears to be unlikely. Based on asking for just these ALB and EBS
+metrics:
+
+```
+ApplicationELB/RequestCount
+ApplicationELB/TargetResponseTime
+ApplicationELB/HTTPCode_ELB_5XX_Count
+EBS/VolumeWriteBytes
+EBS/VolumeReadBytes
+EBS/VolumeReadOps
+EBS/VolumeWriteOps
+```
+
+we appear to be requesting roughly 4000 metrics per scrape. If we scraped our current config at a period of once every 10 minutes we would end up roughly $21,000 a year. If we scraped it every 30 seconds it would
+be about $420,000.
+
+By requesting only ALB metrics in the dev accounts, we still produce about 160 API requests according to the `cloudwatch_requests_total` counter for each scrape. Somewhat strangely, this only returns about 30 timeseries so we are not sure if our config is incorrect and if the number of API calls could be reduced to closer
+match the number of timeseries.
+
+Note, as dev accounts have lots of resources e.g. volumes, there may be fewer
+metrics requested for staging and prod as unlike the dev account we wouldn't be
+exporting metrics for other stacks.
+
+It takes a long time to get a response from the /metrics endpoint as it needs to make many API calls. This causes slow response times for which our prometheus scrape config needs to allow for using the scrape_timeout setting
+
+For speed of development, we still are only running one instance per ECS cluster. There were questions raised about having such as CPU intensive process (that was prone to crashing due to CPU spikes) running on the same instance as prometheus. However we didn't do anything about this as it would require a lot of work to have more than one instance per cluster.
```

Comment:
this seems to assume that the reader knows that cloudwatch_exporter is a CPU-intensive process. Can we maybe have a little context about this?